### ISSUE TITLE
INI, IND, INIR, INDR are incorrect.

### ISSUE THREAD

**Issue by AntonyDTA** [simonowen/simcoupe#75]
Unlike OUTI, OUTD, OTIR and OTDR, register B decrement is performed only after reading from the port.
```
               DUMP 4,0
               ORG  &8000

Test:          LD   BC,&00F8
               LD   HL,Mem
               HALT
               INI
               DEC  HL
               LD   A,(HL)
               AND  &F0            ;&C0=(HPEN)=Bad, &00=(LPEN)=OK
Loop:          OUT  (&F8),A        ;Palette0=Green=Bad, Palette0=Black=OK
               JR   Loop

Mem:           DEFB 0
```

**Comment by simonowen** [comment]
This seems to have been broken since replacing the Z80 CPU core in eac8e8bd578bde8126128119f1da17908c41d3ed about 9 months ago. I fixed B being decremented too _late_ in the block OUT instructions but hadn't noticed that the block IN were doing it too _early_ too.

Thanks for the report!  I'll see if I can create a new build that includes it over the weekend.

**Comment by simonowen** [comment]
Fixed in new [v1.2.9 release](https://github.com/simonowen/simcoupe/releases/tag/v1.2.9).
